lib/Dialect/TritonGPU/Transforms/Pipeliner/Partition.cpp

```diff
@@ -114,6 +114,30 @@ const WarpSchedule::Partition *WarpSchedule::getPartition(unsigned idx) const {
   return partitions[idx].get();
 }
 
+void assignPartitionToBlock(
+    mlir::Block *block, WarpSchedule::Partition *part,
+    DenseMap<Operation *, WarpSchedule::Partition *> &opToPartition);
+
+void assignPartitionToIfOp(
+    scf::IfOp ifOp, WarpSchedule::Partition *part,
+    DenseMap<Operation *, WarpSchedule::Partition *> &opToPartition) {
+  assignPartitionToBlock(ifOp.thenBlock(), part, opToPartition);
+  if (ifOp.elseBlock()) {
+    assignPartitionToBlock(ifOp.elseBlock(), part, opToPartition);
+  }
+}
+
+void assignPartitionToBlock(
+    mlir::Block *block, WarpSchedule::Partition *part,
+    DenseMap<Operation *, WarpSchedule::Partition *> &opToPartition) {
+  for (auto &op : block->getOperations()) {
+    opToPartition[&op] = part;
+    if (auto ifOp = dyn_cast<scf::IfOp>(op)) {
+      assignPartitionToIfOp(ifOp, part, opToPartition);
+    }
+  }
+}
```

Comment:
why do we need to assign a partition to operations within the If region? The region should get moved along with the parent op

Reply:
Because `WarpSchedule` expects all ops to which it calls `getPartition` to have a partition assigned - it uses `DenseMap::at`. 

Consider this lit test: https://github.com/triton-lang/triton/blob/main/test/TritonGPU/partition-loops.mlir#L224-L232
The users of the block argument are in then / else branches. Due to the change in this PR from https://github.com/triton-lang/triton/pull/6395#discussion_r2029678809, all users of a block argument will get called `getPartition` on it, so it must belong to a partition.

Reply:
Done, yeah it works